src/transformState.js

```diff
@@ -5,7 +5,23 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (let i = 0; i < actions.length; i++) {
+    switch (actions[i].type) {
+      case 'addProperties':
+        Object.assign(state, actions[i].extraData);
+        break;
+
+      case 'removeProperties':
+        for (const keys of actions[i].keysToRemove) {
+          delete state[keys];
+        }
+        break;
+
+      case 'clear':
+        Object.keys(state).forEach(key => delete state[key]);
+        break;
+    }
```

Comment:
also is a good point to add default case to switch
it's crucial for handling errors

Reply:
https://stackoverflow.com/questions/4649423/should-switch-statements-always-contain-a-default-clause